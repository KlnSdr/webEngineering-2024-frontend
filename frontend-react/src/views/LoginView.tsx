import { Button } from "react-bootstrap";
import { UserService } from "../services/UserService";

/**
 * Component for the login view.
 * @returns {JSX.Element} The rendered component.
 */
function LoginView(): JSX.Element {
  return (
    <div
      className="d-flex justify-content-center mt-5"
      style={{ marginTop: "25%" }}
    >
      <div className="card">
        <div className="card-header">Anmelden</div>
        <div className="card-body">
          <p className="card-text">
            Um alle Features verwenden zu können müssen Sie sich anmelden.
          </p>
          <div className="btn-group" role="group">
            <Button onClick={UserService.doLogin}>
              <i className="bi bi-gitlab"></i> Login mit GitLab
            </Button>
          </div>
        </div>
      </div>
    </div>
  );
}

export default LoginView;