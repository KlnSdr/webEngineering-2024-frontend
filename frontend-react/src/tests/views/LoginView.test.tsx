import React from "react";
import { render } from "@testing-library/react";
import LoginView from "../../views/LoginView";

describe("LoginView Component", () => {

    it("matches the snapshot", () => {
        const { asFragment } = render(<LoginView />);
        expect(asFragment()).toMatchSnapshot();
    });
});
