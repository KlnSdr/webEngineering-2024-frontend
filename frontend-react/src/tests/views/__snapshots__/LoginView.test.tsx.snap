// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center mt-5"
    style="margin-top: 25%;"
  >
    <div
      class="card"
    >
      <div
        class="card-header"
      >
        Anmelden
      </div>
      <div
        class="card-body"
      >
        <p
          class="card-text"
        >
          Um alle Features verwenden zu können müssen Sie sich anmelden.
        </p>
        <div
          class="btn-group"
          role="group"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            <i
              class="bi bi-gitlab"
            />
             Login mit GitLab
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
